tools/crm_resource_print.c

```diff
@@ -158,41 +158,61 @@ agent_status_default(pcmk__output_t *out, va_list args) {
     const char *provider = va_arg(args, const char *);
     const char *type = va_arg(args, const char *);
     int rc = va_arg(args, int);
+    const char *exit_reason = va_arg(args, const char *);
 
     if (status == PCMK_EXEC_DONE) {
-        out->info(out, "Operation %s%s%s (%s%s%s:%s) returned: '%s' (%d)",
-                  action, name ? " for " : "", name ? name : "",
-                  class, provider ? ":" : "", provider ? provider : "", type,
-                  services_ocf_exitcode_str(rc), rc);
+        /* Operation <action> [for <resource>] (<class>[:<provider>]:<agent>)
+         * returned <exit-code> (<exit-description>[: <exit-reason>])
+         */
+        out->info(out, "Operation %s%s%s (%s%s%s:%s) returned %d (%s%s%s)",
+                  action,
+                  ((name == NULL)? "" : " for "), ((name == NULL)? "" : name),
+                  class, (provider? ":" : ""), (provider? provider : ""), type,
```

Comment:
All the other comparisons are something like `name == NULL` or `exit_reason == NULL`, but these two leave the NULL part off.  It'd be nice to be consistent with one or the other.  This also happens in the else branch of this conditional.

Reply:
Yep, I prefer the == NULL construction because it makes it obvious it's a pointer, and having the empty string case first often makes things line up better